tidy3d/plugins/mode/mode_solver.py

```diff
@@ -348,6 +366,422 @@ def data_raw(self) -> ModeSolverData:
+    def bend_axis_3d(self) -> Axis:
+        """Transform the 2D bend axis to its corresponding 3D axis."""
+        _, idx_plane = self.plane.pop_axis((0, 1, 2), axis=self.normal_axis)
+        return idx_plane[self.mode_spec.bend_axis]
+
+    @cached_property
+    def rotated_mode_solver_data(self) -> ModeSolverData:
+        # Create a mode solver with rotated geometries for a reference solution with 0-degree angle
+        solver_ref = self.rotated_structures_copy
+        solver_ref_data = solver_ref.data_raw
+
+        # The reference data should always be colocated to convert to cylindrical coordinates
+        n_complex = solver_ref_data.n_complex
+        if not solver_ref.colocate:
+            solver_ref_data = solver_ref._colocate_data(mode_solver_data=solver_ref_data)
+
+        # Transform the colocated mode solution from cartesian to cylindrical coordinates
+        solver_ref_data_cylindrical = self._car_2_cyn(mode_solver_data=solver_ref_data)
+
+        try:
+            solver = self.reduced_simulation_copy
+        except Exception as e:
+            solver = self
+            log.warning(
+                "Mode solver reduced_simulation_copy failed. "
+                "Falling back to non-reduced simulation, which may be slower. "
+                f"Exception: {str(e)}"
+            )
+
+        # Compute the mode solution by rotating the reference data to the monitor plane
+        rotated_mode_fields = self._mode_rotation(
+            solver_ref_data_cylindrical=solver_ref_data_cylindrical,
+            solver=solver,
+        )
+
+        # TODO: At a later time, we should ensure that `eps_spec` is automatically returned to
+        # to compute the backward propagation mode solution using a mode solver
+        # with direction "-".
+        eps_spec = []
+        for _ in self.freqs:
+            eps_spec.append("tensorial_complex")
+        # finite grid corrections
+        grid_factors = solver._grid_correction(
+            simulation=solver.simulation,
+            plane=solver.plane,
+            mode_spec=solver.mode_spec,
+            n_complex=n_complex,
+            direction=solver.direction,
+        )
+
+        # Make mode solver data on the Yee grid
+        mode_solver_monitor = solver.to_mode_solver_monitor(name=MODE_MONITOR_NAME, colocate=False)
+        grid_expanded = solver.simulation.discretize_monitor(mode_solver_monitor)
+        rotated_mode_data = ModeSolverData(
+            monitor=mode_solver_monitor,
+            symmetry=solver.simulation.symmetry,
+            symmetry_center=solver.simulation.center,
+            grid_expanded=grid_expanded,
+            grid_primal_correction=grid_factors[0],
+            grid_dual_correction=grid_factors[1],
+            eps_spec=eps_spec,
+            **rotated_mode_fields,
+        )
+
+        # Colocate to grid boundaries if requested
+        if self.colocate:
+            rotated_mode_data = self._colocate_data(mode_solver_data=rotated_mode_data)
```

Comment:
If I'm reading `_mode_rotation` correctly, this can be skipped because the fields have already been created either colocated or not depending on `self.colocate`.